### DISCUSSION TITLE
Permission errors for deno install npm:PACKAGE

### DISCUSSION THREAD

**Discussion by Brixy** [denoland/deno#16924] [Q&A] [answered]
```
deno: 1.28.3
OS: Void Linux
```

With Deno 1.28 I would like to drop node completely.

While using npm modules in my scripts works perfectly installing npm executables (in the home folder) causes permission errors.

This is what I am trying to do:

```
deno install --root ~/.deno -A npm:PACKAGE
```

Error message:

> error: Permission denied (os error 13)

Installation does work with:

```
sudo deno install --root ~/.deno -A npm:PACKAGE
```

Yet, when you run

```
PACKAGE
```

you get this error message:

> error: Uncaught TypeError: Permission denied (os error 13)

```
sudo PACKAGE
```

does work correctly but as the script is called by other scripts it is not always possible to prepend `sudo`.

(The same package installed via node does not require `sudo`.)

Is is possible to run (and possibly install) `PACKAGE` without sudo? Thank you very much.

**Comment by bartlomieju** [comment]
@dsherret please take a look

**Comment by Brixy** [comment, marked as answer]
Installation now works perfectly with deno 1.29.2. No OS permission errors any more.

I’ll mark this question as answered.

---

Stand-alone scripts work fine from the command line (e.g. `mjml` or `mammoth`).

Scripts that require other packages (e.g. `stylelint`) do not work correctly; but this is a different topic.

**Reply by salemalem** [reply]
![image](https://github.com/denoland/deno/assets/42411937/0dae999d-7e9d-4732-bcec-a95b861cf321)
doesn't work on 
deno 1.40.3 (release, x86_64-unknown-linux-gnu)
v8 12.1.285.6
typescript 5.3.3
